meilisearch-lib/src/index_controller/mod.rs

```diff
@@ -0,0 +1,508 @@
+
+        for (index_uid, index) in self.index_resolver.list().await? {
+            let uuid = index.uuid;
+            let (mut stats, meta) = spawn_blocking::<_, IndexResult<_>>(move || {
+                let stats = index.stats()?;
+                let meta = index.meta()?;
+                Ok((stats, meta))
+            })
+            .await??;
+
+            database_size += stats.size;
+
+            last_update = last_update.map_or(Some(meta.updated_at), |last| {
+                Some(last.max(meta.updated_at))
+            });
+
+            // Check if the currently indexing update is from our index.
+            stats.is_indexing = Some(Some(uuid) == update_infos.processing);
+
+            indexes.insert(index_uid, stats);
+        }
+
+        Ok(Stats {
+            database_size,
+            last_update,
+            indexes,
+        })
+    }
+
+    pub async fn create_dump(&self) -> Result<DumpInfo> {
+        Ok(self.dump_handle.create_dump().await?)
+    }
+
+    pub async fn dump_info(&self, uid: String) -> Result<DumpInfo> {
+        Ok(self.dump_handle.dump_info(uid).await?)
+    }
+
+    pub async fn create_index(
+        &self,
+        uid: String,
+        primary_key: Option<String>,
+    ) -> Result<IndexMetadata> {
+        let index = self
+            .index_resolver
+            .create_index(uid.clone(), primary_key)
+            .await?;
+        let meta = spawn_blocking(move || -> IndexResult<_> {
+            let meta = index.meta()?;
+            let meta = IndexMetadata {
+                uuid: index.uuid,
+                uid: uid.clone(),
+                name: uid,
+                meta,
+            };
+            Ok(meta)
+        })
+        .await??;
+
+        Ok(meta)
+    }
+
+    pub async fn delete_index(&self, uid: String) -> Result<()> {
+        let uuid = self.index_resolver.delete_index(uid).await?;
+
+        let update_sender = self.update_sender.clone();
+        tokio::spawn(async move {
+            let _ = UpdateMsg::delete(&update_sender, uuid).await;
+        });
+
+        Ok(())
+    }
+}
+
+pub async fn get_arc_ownership_blocking<T>(mut item: Arc<T>) -> T {
+    loop {
+        match Arc::try_unwrap(item) {
+            Ok(item) => return item,
+            Err(item_arc) => {
+                item = item_arc;
+                sleep(Duration::from_millis(100)).await;
+                continue;
+            }
+        }
+    }
+}
+
+/// Parses the v1 version of the Asc ranking rules `asc(price)`and returns the field name.
+pub fn asc_ranking_rule(text: &str) -> Option<&str> {
+    text.split_once("asc(")
+        .and_then(|(_, tail)| tail.rsplit_once(")"))
+        .map(|(field, _)| field)
+}
+
+/// Parses the v1 version of the Desc ranking rules `asc(price)`and returns the field name.
+pub fn desc_ranking_rule(text: &str) -> Option<&str> {
+    text.split_once("desc(")
+        .and_then(|(_, tail)| tail.rsplit_once(")"))
+        .map(|(field, _)| field)
+}
+
```

Comment:
Where is this code used? Shouldn't it only be used in `dump/extractor/v1.rs`?

Reply:
this is also used in index/dump.rs